### ISSUE TITLE
DefaultCredentialsError: Could not automatically determine credentials.

### ISSUE THREAD

**Issue by colincadams** [Recidiviz/recidiviz-data#511]
We see this error periodically on appengine. It really shouldn't occur afaik, because app engine should always have our credentials.

sample stack trace:
```
Traceback (most recent call last):
  File "/env/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/env/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/env/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/env/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/env/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/env/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/srv/recidiviz/utils/auth.py", line 98, in auth_and_call
    return func(*args, **kwargs)
  File "/srv/recidiviz/ingest/worker.py", line 89, in work
    result = scraper_task(params)
  File "/srv/recidiviz/ingest/base_scraper.py", line 170, in _generic_scrape
    self.add_task('_generic_scrape', task_params)
  File "/srv/recidiviz/ingest/scraper.py", line 304, in add_task
    'params': params
  File "/srv/recidiviz/ingest/queues.py", line 80, in create_task
    queue_path = format_queue_path(queue_name)
  File "/srv/recidiviz/ingest/queues.py", line 46, in format_queue_path
    full_queue_path = client().queue_path(
  File "/srv/recidiviz/ingest/queues.py", line 30, in client
    _client = tasks_v2beta3.CloudTasksClient()
  File "/env/lib/python3.7/site-packages/google/cloud/tasks_v2beta3/gapic/cloud_tasks_client.py", line 190, in __init__
    address=self.SERVICE_ADDRESS, channel=channel, credentials=credentials
  File "/env/lib/python3.7/site-packages/google/cloud/tasks_v2beta3/gapic/transports/cloud_tasks_grpc_transport.py", line 60, in __init__
    channel = self.create_channel(address=address, credentials=credentials)
  File "/env/lib/python3.7/site-packages/google/cloud/tasks_v2beta3/gapic/transports/cloud_tasks_grpc_transport.py", line 84, in create_channel
    address, credentials=credentials, scopes=cls._OAUTH_SCOPES
  File "/env/lib/python3.7/site-packages/google/api_core/grpc_helpers.py", line 177, in create_channel
    credentials, _ = google.auth.default(scopes=scopes)
  File "/env/lib/python3.7/site-packages/google/auth/_default.py", line 306, in default
    raise exceptions.DefaultCredentialsError(_HELP_MESSAGE)
google.auth.exceptions.DefaultCredentialsError: Could not automatically determine credentials. Please set GOOGLE_APPLICATION_CREDENTIALS or explicitly create credentials and re-run the application. For more information, please see https://cloud.google.com/docs/authentication/getting-started
```

Seems like a common issue 😢 
https://github.com/googleapis/google-auth-library-python/issues/211

**Comment by arian487** [comment]
Do we still see this error?  I haven't seen it myself.

**Comment by colincadams** [comment]
yep: https://pantheon.corp.google.com/errors/CKzFrvyv5cbCugE?time=P7D&project=recidiviz-123

**Comment by colincadams** [comment]
It looks like this may have been fixed by #1332. It is hard to tell, as batch persistence still read tasks that failed with this error last night, but it looks like these were all old tasks. Not all of the pubsub queues were cleared but with #1350 tonight there should only be new tasks in the queue, so we should be able to find out for sure.

**Comment by colincadams** [comment]
This hasn't shown up in the last few days.
